service/dap/server.go

```diff
@@ -721,18 +729,53 @@ func (s *Server) onScopesRequest(request *dap.ScopesRequest) {
 	// Retrieve local variables
 	locals, err := s.debugger.LocalVariables(scope, cfg)
 	if err != nil {
-		s.sendErrorResponse(request.Request, UnableToListLocals, "Unable to list local vars", err.Error())
+		s.sendErrorResponse(request.Request, UnableToListLocals, "Unable to list locals", err.Error())
 		return
 	}
 	locScope := api.Variable{Name: "Locals", Children: locals}
 
 	// TODO(polina): Annotate shadowed variables
-	// TODO(polina): Retrieve global variables
 
 	scopeArgs := dap.Scope{Name: argScope.Name, VariablesReference: s.variableHandles.create(argScope)}
 	scopeLocals := dap.Scope{Name: locScope.Name, VariablesReference: s.variableHandles.create(locScope)}
 	scopes := []dap.Scope{scopeArgs, scopeLocals}
 
+	if s.args.loadGlobalVariables {
+		state, err := s.debugger.State( /*nowait*/ true)
+		if err != nil {
+			s.sendErrorResponse(request.Request, UnableToListGlobals, "Unable to list globals", err.Error())
+			return
+		}
+
+		// List global variables only for the current package only
+		currPkg := "??"
+		var globals []api.Variable
+		if state.CurrentThread != nil && state.CurrentThread.File != "" {
+			currDir := path.Dir(state.CurrentThread.File)
+			pkgInfo := s.debugger.ListPackagesBuildInfo( /*includeFiles*/ false)
+			for _, pkg := range pkgInfo {
```

Comment:
Could you please add a comment here explaining what's going on? What are these path manipulations doing, in terms of what the debugger returns vs. what we want to send back on DAP

Reply:
Added commnents.